Services/AspireService.cs

```diff
@@ -1,93 +1,163 @@
+                    return;
+                }
+
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                if (type.Equals(WatchEventType.Added) || type.Equals(WatchEventType.Modified))
+                {
+                    watchResourcesUpdate.Changes.Value.Add(new WatchResourcesChange() { Upsert = { Name = appName, DisplayName = appName, ResourceType = item.Kind, Uid = item.Uid(), CreatedAt = Google.Protobuf.WellKnownTypes.Timestamp.FromDateTime(item.CreationTimestamp().Value), State = item.Status.Phase, Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = item.Status.PodIP } }, Properties = { new ResourceProperty() { DisplayName = item.Kind, Name = item.Metadata.Name } } } });
+                    await responseStream.WriteAsync(watchResourcesUpdate);
+                }
+                else if (type.Equals(WatchEventType.Deleted))
+                {
+                    watchResourcesUpdate.Changes.Value.Add(new WatchResourcesChange() { Delete = new ResourceDeletion { ResourceName = item.Metadata.Name, ResourceType = item.Kind } });
+                    await responseStream.WriteAsync(watchResourcesUpdate);
+                }
+            });
+        }
+
+        public override async Task WatchResourceConsoleLogs(WatchResourceConsoleLogsRequest request, IServerStreamWriter<WatchResourceConsoleLogsUpdate> responseStream, ServerCallContext context)
+        {
+            var label = $"containerapps.io/app-name={request.ResourceName}";
+            _logger.LogInformation($"Label : {label}");
+            var pods = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace, labelSelector: label);
+
+            // We only have one pod deployed for each Aspire App
+            if (pods.Items.Count() > 1)
+            {
+                _logger.LogDebug($"Multiple pods were found for the app {request.ResourceName}");
+            }
+
+            var podName = pods.Items[0].Metadata.Name;
+            var container = pods.Items[0].Spec.Containers.Where(c => c.Name == request.ResourceName).FirstOrDefault();
+            if (container == null)
+            {
+                _logger.LogWarning($"Container matching {request.ResourceName} does not exist");
+                return;
+            }
+            
+            _logger.LogInformation($"PodName: {podName}");
+            var stream = new System.IO.MemoryStream();
+            while (!context.CancellationToken.IsCancellationRequested)
+            {
+                // Re-set stream to the beginning to make sure repetitive logs are not read. 
+                stream.Seek(stream.Length, SeekOrigin.Begin);
+                stream = (MemoryStream)await _kubernetesClient.CoreV1.ReadNamespacedPodLogAsync(podName, _kubernetesConfig.Namespace, container: container.Name, follow: true);

```

Comment:
I see that we have this `follow: true` argument, but it doesn't feel like we're doing anything with it? Am I missing something?

Reply:
We are not doing anything with this. I copied this line from the doc, I'll remove it if this does not make sense